services/web-app/components/catalog/catalog.js

```diff
@@ -51,37 +66,155 @@ const assetIsInFilter = (asset, filter) => {
   return true
 }
 
+/**
+ * Sorts and filters an array of assets given a filter, sort key, and search query. Until a better
+ * solution is in place, the search is simply a filter to remove assets that don't match any part of
+ * the name or description.
+ * @param {import('../../typedefs').Asset[]} assets
+ * @param {Object} filter
+ * @param {string} sort
+ * @param {string} search
+ * @returns {import('../../typedefs').Asset[]}
+ */
+const getFilteredAssets = (assets, filter, sort, search) => {
+  return assets
+    ? [...assets]
+        .sort(assetSortComparator(sort))
+        .filter((asset) => assetIsInFilter(asset, filter))
+        .filter((asset) => {
+          const { description = '', name = '' } = asset.content
+
+          if (search) {
+            return (
+              (name && name.toLowerCase().includes(search.toLowerCase())) ||
+              (description && description.toLowerCase().includes(search.toLowerCase()))
+            )
+          }
+
+          return true
+        })
+    : []
+}
+
+/**
+ * Checks if the value of a filter property is valid. Acceptance criteria:
+ * - Has a value
+ * - Value is of Array type
+ * - The property key is defined in `filters` object and its values are defined
+ * - Each entry in the parametered `value` is contained in the list of acceptable values for the
+ * key as defined in `filter`
+ * @param {Object} filter
+ * @param {string} key
+ * @param {string[]} value
+ * @returns {boolean} True if value is valid
+ */
+const filterPropertyHasValidValue = (filter, key, value) => {
+  return (
+    !!value &&
+    value.constructor === Array &&
+    !!filter?.[key]?.values &&
+    !value.some((val) => !Object.keys(filter[key].values).includes(val))
+  )
+}
+
 function Catalog({ collection, data, type, filter: defaultFilter = {}, glob = {} }) {
-  const [query, setQuery] = useQueryState('q', {
-    defaultValue: ''
-  })
+  const [possibleFilterValues, setPossibleFilterValues] = useState(getFilters({ collection, type }))
+
+  const [query, setQuery] = useQueryState(
+    'q',
+    {
+      defaultValue: ''
+    },
+    (value) => !!value
+  )
 
   const [search, setSearch] = useState(query)
 
-  const [sort, setSort] = useQueryState('sort', {
-    defaultValue: 'a-z',
-    saveToStorage: true
-  })
+  const [sort, setSort] = useQueryState(
+    'sort',
+    {
+      defaultValue: 'a-z',
+      saveToStorage: true
+    },
+    (value) => {
+      return !!value && typeof value === 'string' && ['a-z', 'status'].includes(value)
```

Comment:
`value` will be coerced to a boolean here automatically, so no need to include the `!!`.

Reply:
Done!